endpoints/partition_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package endpoints_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/aws-sdk-go-base/v2/endpoints"
+)
+
+func TestDefaultPartitions(t *testing.T) {
+	t.Parallel()
+
+	got := endpoints.DefaultPartitions()
+	if len(got) == 0 {
+		t.Fatalf("expected partitions, got none")
+	}
+}
+
+func TestPartitionForRegion(t *testing.T) {
+	t.Parallel()
+
+	testcases := map[string]struct {
+		expectedFound bool
+		expectedID    string
+	}{
+		"us-east-1": {
+			expectedFound: true,
+			expectedID:    "aws",
+		},
+		"us-gov-west-1": {
+			expectedFound: true,
+			expectedID:    "aws-us-gov",
+		},
+		"not-found": {
+			expectedFound: false,
+		},
+		"us-east-17": {
```

Comment:
This one was surprising to me. Is there a reason `PartitionForRegion` matches on the regex rather than comparing to the list of region name literals?

Not a blocker, just curious about the history here.

Reply:
I kept that logic for backwards compatibility with AWS SDK for Go v1:

> // PartitionForRegion returns the first partition which includes the region
> // passed in. This includes both known regions and regions which match
> // a pattern supported by the partition which may include regions that are
> // not explicitly known by the partition. Use the Regions method of the
> // returned Partition if explicit support is needed.